torch/fx/experimental/symbolic_shapes.py

```diff
@@ -363,24 +387,23 @@ def _maybe_guard_eq(self, expr: "sympy.Eq") -> None:
         lhs = expr.lhs
         rhs = expr.rhs
         try:
-            solutions = sympy.solveset(lhs - rhs, free[0], domain=sympy.S.Integers)
-            if not solutions.is_finite_set:
-                if expr.has(sympy.Mod):
-                    mod_expr = tuple(expr.atoms(sympy.Mod))[0]
-                    solutions = sympy.solveset(lhs - rhs, mod_expr, domain=sympy.S.Integers)
```

Comment:
It feels like Integers domain should have been OK here; how come it wasn't fine?

Reply:
`solveset` is actually apparently slower 🤔 than `solve` (according to developer). Also, `solveset` ended up having a bunch of annoying errors, and generally seems less robust than `solve`.